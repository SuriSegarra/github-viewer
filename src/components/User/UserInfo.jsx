import React from 'react';
import PropTypes from 'prop-types';

const UserInfo = ({ login, followers, following, url, avatar }) => (
  <figure>
    <img src={avatar} alt={login}/>
    <figcaption>
      <h1>{login}</h1>
      <p>Followers:{followers}</p>
      <p>Following:{following}</p>
      <p>Profile:{url}</p>
    </figcaption>
  </figure>
);

UserInfo.propTypes = {
  login: PropTypes.string.isRequired,
  followers: PropTypes.string.isRequired,
  following: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  avatar: PropTypes.string.isRequired

};

export default UserInfo;
