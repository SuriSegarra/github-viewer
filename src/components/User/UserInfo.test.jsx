import React from 'react';
import { shallow } from 'enzyme';
import UserInfo from './UserInfo';

describe('UserInfo', () => {
  it('matches a userInfo snapshot', () => {
    const wrapper = shallow(<UserInfo login='Surelis Segarra' 
      followers='1' 
      following='3' 
      url='https://github.com/SuriSegarra' 
      avatar='https://avatars2.githubusercontent.com/u/57576258?v=4' />);
    expect(wrapper).toMatchSnapshot();
  });
});
