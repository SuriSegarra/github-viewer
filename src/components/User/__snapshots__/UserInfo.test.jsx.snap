// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo matches a userInfo snapshot 1`] = `
<figure>
  <img
    alt="Surelis Segarra"
    src="https://avatars2.githubusercontent.com/u/57576258?v=4"
  />
  <figcaption>
    <h1>
      User:
      Surelis Segarra
    </h1>
    <p>
      Followers:
      1
    </p>
    <p>
      Following:
      3
    </p>
    <p>
      Profile:
      https://github.com/SuriSegarra
    </p>
  </figcaption>
</figure>
`;
